src/core/common-functions/execCommands.ts

```diff
@@ -0,0 +1,50 @@
+import { execSync } from 'child_process';
+import { env } from 'process';
+import { AppLogger } from '../../logger/appLogger.js';
+
+type StdioOption = 'inherit' | 'pipe' | 'ignore';
+type ShellOption = string | undefined; 
+interface ExecuteCommandOptions {
+    cwd?: string;
+    maxRetries?: number;
+    stdio?: StdioOption;
+    shell?: ShellOption;
+    env?:ShellOption;
+}
+
+export async function executeCommandWithRetry(
+    command: string,
+    options: ExecuteCommandOptions = {} 
+) {
+    const {
+        cwd = process.cwd(),
+        maxRetries = 3,
+        stdio = 'inherit',
+        shell = undefined
+    } = options;
+
+    let attempts = 0;
+    let success = false;
+
+    while (attempts < maxRetries && !success) {
+        try {
+            attempts++;
+            execSync(command, { cwd, stdio, shell, env });
```

Comment:
Again, here.. we are not checking what will be returned by execSync. We do have to capture it and also take some action based on it..